config/initializers/rack_attack.rb

```diff
@@ -38,6 +38,8 @@ class Rack::Attack
   def self.protected_route?(protected_actions, path, method)
     route_params = Rails.application.routes.recognize_path(path, method: method)
     protected_actions.any? { |hash| hash[:controller] == route_params[:controller] && hash[:action] == route_params[:action] }
+  rescue ActionController::RoutingError # raised for routes with a redirect
+    false if method == "GET" && path.match(%r{/users/(.)*}) # allow /users/<username> to redirect to /profile/<username>
```

Comment:
I feel like the `if` expression is over engineering the problem. If the current route is not recognized, then it's not a protected route and just returning `false` is perfectly fine.

Reply:
are we sure we want to support this at cost of this code complexity? I would lean towards @simi suggestion that typo should be fixed. What if we get another issue where another user had used avatars, accounts, etc instead of profiles?

Reply:
We will need the `rescue` for #2297